src/region/RegionAnnotations.tsx

```diff
@@ -63,7 +62,7 @@ export default class RegionAnnotations extends React.PureComponent<Props, State>
 
     handleStop = (shape: Rect): void => {
         const { location, setStaged, setStatus } = this.props;
-        setStaged({ location, shape });
+        setStaged({ location, shape, type: 'region' });
```

Comment:
Is `type` needed here or can we add it in the action?

Reply:
They `type` is only used for the `CreatorItemHighlight` type -- mainly used for the type guard. If we're having the type guard narrow the type by using specific properties on that type, I can get rid of this `type`. Something like 
```
const isCreatorStagedHighlight = (staged: CreatorItem | null): staged is CreatorItemHighlight => {
    return 'shapes' in staged;
};
```

Reply:
Seems okay to me.